test/lit/passes/directize_all-features.wast

```diff
@@ -807,6 +808,52 @@
    )
   )
  )
+
+ ;; CHECK:      (func $select-non-nullable-unreachable-condition
+ ;; CHECK-NEXT:  (call_indirect $0 (type $F)
+ ;; CHECK-NEXT:   (ref.func $select-non-nullable)
+ ;; CHECK-NEXT:   (select
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:    (i32.const 11)
+ ;; CHECK-NEXT:    (unreachable)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $select-non-nullable-unreachable-condition
+  ;; Test we ignore an unreachable condition and don't make any changes at all
+  ;; to the code (in particular, we shouldn't add any vars).
+  (call_indirect (type $F)
+   (ref.func $select-non-nullable)
+   (select
+    (i32.const 10)
+    (i32.const 11)
+    (unreachable)
+   )
+  )
+ )
+
+ ;; CHECK:      (func $select-non-nullable-unreachable-arg (param $x i32)
+ ;; CHECK-NEXT:  (call_indirect $0 (type $F)
+ ;; CHECK-NEXT:   (unreachable)
+ ;; CHECK-NEXT:   (select
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:    (i32.const 11)
+ ;; CHECK-NEXT:    (local.get $x)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $select-non-nullable-unreachable-arg (param $x i32)
```

Comment:
Did we do anything for this in this PR?

Reply:
Technically the PR didn't do anything for it, correct - this wasn't broken before (since we checked unreachability before adding any vars). Still, for completeness this test means we check all the corner cases in this area to verify we never create a var if we don't optimize.